### ISSUE TITLE
Report which properties are missing when using Extractors

### ISSUE THREAD

**Issue by sveinelo** [hamnis/immutable-json#7]
Currently I can't find a way to make Extractors report which expected fields are missing.

```
Json.JObject json = jObject(entry("present", "value"));
Extractors.extract1(TypedField.TString("expected"), Tuple1::new).apply(json)
  .orElseThrow(() -> new RuntimeException("Some expected property is missing"));
```

**Comment by hamnis** [comment]
The reason for this is that we currently use Optional, which does not have this feature.
I have been thinking about creating something like a Cursor, inspired by `circe` or `argonaut`, but I'm not sure if this is possible, or feasable in Java.

**Comment by hamnis** [comment]
This is party fixed in 3.0.0 by introducing DecodeResult, and made usable in 3.5.0. Now you can extract and get a message that some value failed in an object.
